app/controllers/team_members_controller.rb

```diff
@@ -1,42 +1,50 @@
 class TeamMembersController < ApplicationController
 
-  def edit
-    @team_member = TeamMember.find(params[:id])
+  before_action :set_team_member
 
+  def edit
     if @team_member.role == 'custodian'
       @team_members = @team_member.team.team_members.joins(:user).where.not(users: {full_name: ''} )
     end
   end
 
   def update
-
-    team_member = TeamMember.find(params[:id])
-
-    if team_member.role == 'custodian'
+    if @team_member.role == 'custodian'
       if params[:new_custodian_team_member_id] == params[:id]
         flash[:notice] = 'Your custodian remains unchanged'
       else
         new_custodian = TeamMember.find(params[:new_custodian_team_member_id])
 
-        team_member.role = 'advanced'
+        @team_member.role = 'advanced'
         new_custodian.role = 'custodian'
 
-        team_member.save
+        @team_member.save
         new_custodian.save
 
         flash[:notice] = 'The custodian has been changed'
       end
-      redirect_to team_path(team_member.team_id)
+      redirect_to team_path(@team_member.team_id)
     else
-      if team_member.update(role: params[:team_member][:role])
+      if @team_member.update(role: params[:team_member][:role])
         flash[:notice] = 'Your update has been successful'
-        redirect_to team_path(team_member.team_id)
+        redirect_to team_path(@team_member.team_id)
       else
         flash[:notice] = 'Your update has failed'
         render :edit
       end
     end
+  end
+
+  def destroy
+    @team_member.destroy
+    flash[:notice] = "Successfully removed #{@team_member.user.full_name} from team"
+    redirect_to team_path(@team_member.team_id)
+  end
 
+  private
+
+  def set_team_member
```

Comment:
"Standard convention" and common convention are 2 different things. Rails don't have that pattern in their docs from what I can see so its not standard convention.

So dry is part of the clean code principals where it says code
3. It should not be redundant

General rule of thump on dry from working in industry is you should only consider making a method for something if its repeated 3+ times as otherwise its not worth it.

and code according to clean code should also be
4. Reading your code should be pleasant
6. It should have minimal dependencies
9. It should be expressive

i.e. you're adding 3 lines to remove 2 which makes the code less readable, hides where the variable is coming from makes it harder to read, makes it less expressive and creates dependency. 

There are situations that make the pattern useful I am sure but its not this one.